hack/verify-test-featuregates.sh

```diff
@@ -41,19 +41,20 @@ if [[ -n "${direct_sets}" ]]; then
   rc=1
 fi
 
+function verify_no_new_unversioned_feature_gates {
+  local new_features_file=$1
+  local old_features_file="__masterbranch/${new_features_file}"
+  mkdir -p "$(dirname "${old_features_file}")"
+  git show master:"${new_features_file}" > "${old_features_file}"
```

Comment:
Can we always rely on git master being a clean representation of the tree minus what a contributor has changed?  Should we consider instead maintaining an allow list of old unversioned features and using that instead of diff-ing against master?

Reply:
I think the git master should be a clean representation of the tree if all new feature development goes into master. This is just more generic than maintaining multiple lists of old features for different components.

Reply:
Something I often do:

- Pull master
- Create branch A, do some development
- Decide I need to do something else for a while
- Checkout master
- Pull master
- Create branch B, do some development
- Find I need to finish something on branch A
- Checkout branch A
- (is master still a clean representation of branch A minus my development work?)

Reply:
when you 
* Checkout branch A for the 2nd time, the master could have stayed the same, or some old feature `FeatureA` have been removed. In case of master feature removal, you will see verify failure with the message of the `FeatureA` should not be added to the unversioned map. If you know you did not add `FeatureA` that should mean it is time to rebase. 

Verify should really be blocking if you are ready to merge. If you just want to grab someone else's branch to check it out, you can just ignore the verify, isn't it?

Reply:
> If you know you did not add FeatureA that should mean it is time to rebase.

I expect this to be confusing to contributors.  I think we should avoid getting git involved in this.  We've had similar problems in the past with git comparisons and eventually migrated away from that approach to more hermetic/repeatable linter processes.

> Verify should really be blocking if you are ready to merge. If you just want to grab someone else's branch to check it out, you can just ignore the verify, isn't it?

Or maybe I'm going to use their branch for feature development... people use git in all sorts of ways..  Having a script that only works when people use git the way we expect them to use it seems prone to problems down the road.